zebrad/tests/acceptance.rs

```diff
@@ -1052,3 +1055,159 @@ async fn tracing_endpoint() -> Result<()> {
+/// exclusive use of the port. The second node will panic with the Zcash metrics
+/// conflict hint added in #1535.
+#[test]
+fn zcash_metrics_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created metrics endpoint_addr
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened metrics endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `metrics.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same tracing listener port, but different
+/// state directories and Zcash listener ports. The first node should get
+/// exclusive use of the port. The second node will panic with the Zcash tracing
+/// conflict hint added in #1535.
+#[test]
+fn zcash_tracing_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created tracing endpoint_addr
+    let mut config = default_test_config()?;
+    config.tracing.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened tracing endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `tracing.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same state directory, but different Zcash
+/// listener ports. The first node should get exclusive access to the database.
+/// The second node will panic with the Zcash state conflict hint added in #1535.
+#[test]
+fn zcash_state_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // A persistent config has a fixed temp state directory, but asks the OS to
+    // automatically choose an unused port
+    let mut config = persistent_test_config()?;
+    let dir_conflict = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(
+        dir_conflict.path(),
+        "Opened Zebra state cache",
```

Comment:
Maybe we should check the path here as well?

```suggestion
        format!("Opened Zebra state cache at {:?}", dir_conflict),
```